trackhubs/models.py

```diff
@@ -268,6 +273,193 @@ def get_trackdb_file_type_count(self):
+        elif species_scientific_name.lower() == 'danio_rerio':
+            if assembly_name.lower() == 'grcz10':
+                division = 'www'
+            elif assembly_name.lower() == 'zv9':
+                division = 'mar2015.archive'
+
+        # if it's fruitfly assembly must take archive into account
+        elif species_scientific_name.lower() == 'drosophila_melanogaster':
+            if assembly_name.lower() == 'release_6':
+                division = 'www'
+            elif assembly_name.lower() == 'release_5':
+                division = 'dec2014.archive'
+
+        # Handle old maize assembly
+        elif species_scientific_name.lower() == 'zea mays':
+            if assembly_name.lower() == 'b73_refgen_v3':
+                division = 'archive.plants'
+            elif assembly_name.lower() == 'agpv4':
+                # AGPv4 is new assembly but doesn't have accession
+                division = 'plants'
+
+        # Handle old wheat assembly, but also new since EG interface doesn't work here
+        elif species_scientific_name.lower() == 'triticum_aestivum':
+            if assembly_name.lower() == 'iwgsc1+popseq':
+                division = 'archive.plants'
+            # TGACv1 is the 'newest' old assembly
+            elif assembly_name.lower() == 'tgac':
+                division = 'oct2017-plants'
+            # IWGSC is new assembly and has accession, but
+            # as said above EG interface cannot fetch entry from genome shared db
+            # this is weird as same code elsewhere using EG interface works
+            elif assembly_name.lower() == 'iwgsc':
+                division = 'plants'
+
+        else:
+            # Normal cases are here
+            # see Translator.pm line 1231 onward
+
+            # Look up division using Ensembl Rest API by using providing the assembly accession
+            assembly_info_url = 'https://rest.ensembl.org/info/genomes/assembly/' + assembly_accession + ''
+            response = requests.get(assembly_info_url, headers={'Accept': 'application/json'}, verify=True)
+            if not response.ok:
+                print("Couldn't get division for assembly '{}', reason: {} [{}]"
+                      .format(assembly_accession, response.text, response.status_code))
+                sys.exit
+
+            # genome_division can be: 'EnsemblVertebrates', 'EnsemblProtists', 'EnsemblMetazoa',
+            # 'EnsemblPlants', 'EnsemblFungi' or 'EnsemblBacteria'
+            # See: https://rest.ensembl.org/documentation/info/info_divisions
+            genome_division = response.json()['division'].lower()
+
+            if genome_division == 'ensemblvertebrates':
+                division = 'www'
+            else:
+                # if genome_division.startswith('ensembl'):
+                # remove 'ensembl' string
+                division = genome_division[len('ensembl'):]
+
+        if division:
+            domain = domain.replace('### DIVISION ###', division)
```

Comment:
Why not just use string format like you do elsewhere?

Reply:
Sorry I did get what you mean here

Reply:
You're right, the simpler the better :)